add centosstream-8

bryceml opened the pull request:
### What does this PR do?
adds centosstream-8

Commit:
add centosstream-8

Ch3LL commented:
re-run full all

bryceml commented:
https://jenkins.saltproject.io/job/pr-centosstream-8-x86_64-py3-pytest/job/PR-60141/4/testReport/tests.integration.cli.test_batch/BatchTest/test_batch_run_grains_targeting/

is the failing test.  Looks like it can't handle the space in `CentOS Stream`

Ch3LL commented:
@bryceml are you looking into the failure or does another engineer need to investigate?

bryceml commented:
it was assigned to @waynew to fix.  If it needs re-assigned that's fine with me.

waynew reviewed (dismissed):
:rocket:

waynew committed:
Use ? for matching spaces

Technically this isn't *quite* right as 'CentOSyStream' would also
match, but it's pretty reasonable:

- OS grains shouldn't ever be that kind of close
- This test is only swapping out spaces, and only for the os grain. That
  would mean there would have to be two OSes with grains that only
  differ by one having a space where another one has any other
  character.
- This test really isn't even about matching grains, we're just using
  compound matching and that's a reasonable one to use.

waynew commented on tests/integration/cli/test_batch.py:
```diff
@@ -58,7 +58,9 @@ def test_batch_run_grains_targeting(self):
         cmd = self.run_salt(
-            '-C "G@os:{} and not localhost" -b 25% test.ping'.format(os_grain),
+            '-C "G@os:{} and not localhost" -b 25% test.ping'.format(
+                os_grain.replace(" ", "?")
```
@bryceml I added this fix to the test that should make things work. We use fnmatch for globbing, and `?` matches any character. While this isn't a comprehensive fix, I think it's good enough. I tried a bunch of other methods:

- quotes: `G@os:"CentOS Stream"`
- backslash:  `G@os:CentOS\ Stream`
- brackets: `G@os:CentOS[ ]Stream`

I probably tried a few others that I'm forgetting now, but I wasn't able to figure out a way to get Salt to recognize any of those as valid, giving me "invalid compound target"

Unless someone else has a better solution, this was the best tradeoff:

- this test isn't actually testing how the OS grain was configured, only that grain targeting works
- for this test to allow a false positive (i.e. pass when it should have failed) we would literally have to have two OS grains configured that differ only by a space vs non-space like CentOS Stream and CentOSxStream.

It's reasonable enough, and the test *should* pass now.

Commit:
update jenkinsfile for debian-11 to match others

Commit:
add amis.yml
